### PR TITLE
Update _brain.py to allow usage of dict in legends

### LINKED ISSUE #11288: add_volume_labels() method from mne.viz.get_brain_class() doesn't allow receiving a dict as legend
### Description of the problem

Hi,

In the MNE documentation for the add_volume_labels() method  from mne.viz.get_brain_class(), it's stated that you pass a dictionary to the legend option, which is actually implemented on the code. However, line 2475 from the _brain.py file actually does a wrong validation since it is set to `_validate_type(legend, (bool, None), 'legend')`; therefore, it actually doesn't allow using the dict option. It should be set to `validate_type(legend, (bool, None, dict), 'legend')` in order to make it work.

Best,
Mauricio

### Steps to reproduce

```Python
This is the code I was using to reproduce:
%matplotlib inline
import os.path as op
import mne
import numpy as np

sample_data_folder = mne.datasets.sample.data_path()
subjects_dir = sample_data_folder / 'subjects'

fname_aseg = op.join(subjects_dir, 'sample', 'mri', 'aseg.mgz')
label_names = mne.get_volume_labels_from_aseg(fname_aseg)

Brain2 = mne.viz.get_brain_class()
brain2 = Brain2('sample', hemi='both', surf='pial', alpha=0.2,
              cortex='low_contrast', subjects_dir=subjects_dir, 
              size=(800, 600))
plotargs = {
    'bcolor':(0.5, 0.5, 0.5), 
    'border':False,
    'size':(0.2, 0.6),
    'loc':'upper left'
}
brain2.add_volume_labels(aseg='aseg', labels=label_names, alpha=1, smooth=0, legend=plotargs)
```

### Link to data

_No response_

### Expected results

The legend is shown in the qt widget.

### Actual results

Legend is not shown and an error is displayed.

### Additional information

Platform:         Linux-5.15.0-52-generic-x86_64-with-glibc2.35
Python:           3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:35:26) [GCC 10.4.0]
Executable:       /home/mcespedes/miniconda3/envs/mne/bin/python
CPU:              x86_64: 8 cores
Memory:           15.4 GB

mne:              1.2.1
numpy:            1.23.4 {OpenBLAS 0.3.21 with 8 threads}
scipy:            1.9.3
matplotlib:       3.6.0 {backend=module://matplotlib_inline.backend_inline}

sklearn:          1.1.3
numba:            0.56.3
nibabel:          4.0.2
nilearn:          0.9.2
dipy:             1.5.0
openmeeg:         Not found
cupy:             Not found
pandas:           1.5.1
pyvista:          0.36.1 {OpenGL 4.6 (Core Profile) Mesa 22.0.5 via Mesa Intel(R) Xe Graphics (TGL GT2)}
pyvistaqt:        0.9.0
ipyvtklink:       0.2.2
vtk:              9.2.2
qtpy:             2.2.1 {PyQt5=5.15.6}
ipympl:           Not found
...
mne_features:     Not found
mne_qt_browser:   0.4.0
mne_connectivity: Not found
mne_icalabel:     Not found

### PR DESCRIPTION
**Pull request by mcespedes99**
Related to the issue #11288 

Thanks for contributing a pull request! Please make sure you have read the
[contribution guidelines](https://mne.tools/dev/install/contributing.html)
before submitting.

Please be aware that we are a loose team of volunteers so patience is
necessary. Assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.

Again, thanks for contributing!

#### Reference issue
Example: Fixes #11288 

#### What does this implement/fix?

In the MNE documentation for the add_volume_labels() method from mne.viz.get_brain_class(), it's stated that you pass a dictionary to the legend option, which is actually implemented on the code. However, line 2475 from the _brain.py file actually does a wrong validation since it is set to _validate_type(legend, (bool, None), 'legend'); therefore, it actually doesn't allow using the dict option. I set it to validate_type(legend, (bool, None, dict), 'legend') in order to make it work.

#### Additional information
It's not possible to change legends of volume labels in figures as it gives an error with the current implementation. In my case, the legend was so small that I couldn't read it. This will allow more custumization for the user.

### CODE REVIEW

**Comment by larsoner** [comment]
With any new change we try to add some test that would fail on `main` but pass on the PR (test-driven development), could you modify the call in:
```
mne/viz/_brain/tests/test_brain.py:334:    brain.add_volume_labels(
```
to pass a `dict`? I think it would fail on `main` and pass on this PR

Then could you add a line to the top of the `Bugs` section of `doc/changes/latest.inc` using `:newcontrib:` and add your name to `doc/changes/names.inc`? Then I think we should be good to merge!

**Comment by mcespedes99** [comment]
Just push that! Let me know if it looks good

**Comment by larsoner** [comment]
Yes that looks great, marking for merge when CIs are happy -- thanks in advance!

**Comment by mcespedes99** [comment]
> Yes that looks great, marking for merge when CIs are happy -- thanks in advance!

Happy to contribute!

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -30,6 +30,7 @@ Enhancements
 
 Bugs
 ~~~~
+- Fix bug to be able to customize legends when using :func:`mne.viz._brain.add_volume_labels` (:gh:`11289` by :newcontrib:`Mauricio Cespedes Tenorio`)
```
Comment (larsoner) [inline doc/changes/latest.inc:33]:
```suggestion
- Fix bug to be able to customize legends when using :meth:`mne.viz.Brain.add_volume_labels` (:gh:`11289` by :newcontrib:`Mauricio Cespedes Tenorio`)
```

**Pull request merged**